internal/pkg/cli/deploy/workload.go

```diff
@@ -428,6 +537,90 @@ func buildArgsPerContainer(name, workspacePath string, img ContainerImageIdentif
 	return dArgs, nil
 }
 
+// copyOutputToBuffer copies the build and push output from the given io.Reader to the buildPushOutputBuffer buffer.
+// return an error if copying fails or if the reader returns an unexpected error.
+func copyOutputToBuffer(pr io.Reader, buffer *buildPushOutputBuffer) error {
+	defer func() {
+		buffer.doneMu.Lock()
+		buffer.done = true
+		buffer.doneMu.Unlock()
+	}()
+
+	// copy the build and push output to the output buffer
+	_, err := io.Copy(buffer, pr)
+	if err == io.EOF {
+		return nil
+	} else if err != nil {
+		return fmt.Errorf("copying build and push output for container %s: %w", buffer.ContainerName, err)
+	}
+	return nil
+}
+
+// printOutputFromBuffers prints the build and push output from the list of buildPushOutputBuffer buffers to stderr.
+// It polls each buffer until all build and push calls are completed,
+// and erases the previous output after sleeping for a short duration.
+func printOutputFromBuffers(buffers []*buildPushOutputBuffer) error {
+	for {
+		writtenLines := 0
+		// check whether all build and push calls are completed.
+		allDone := true
+		for _, buf := range buffers {
+			buf.doneMu.Lock()
+			if !buf.done {
+				allDone = false
+			}
+			buf.doneMu.Unlock()
+			label, logs := buf.logs()
+			outputLogs := removeEmptyStrings(logs)
+			if len(outputLogs) > 0 {
+				fmt.Fprintln(os.Stderr, label)
+				for _, logLine := range outputLogs {
+					fmt.Fprintln(os.Stderr, "\t", logLine)
+				}
+				labelLength, err := dockerBuildLabelLength(label)
+				if err != nil {
+					return fmt.Errorf("get terminal size %w", err)
+				}
+				writtenLines = writtenLines + labelLength + len(outputLogs)
```

Comment:
i wonder if we need this for _every_ line? in case of a docker build line longer than the term. width?